docs/Formula-Cookbook.md

```diff
@@ -750,6 +750,12 @@ ln_s libexec/"name", bin
 
 The symlinks created by [`install_symlink`](https://rubydoc.brew.sh/Pathname#install_symlink-instance_method) are guaranteed to be relative. `ln_s` will only produce a relative symlink when given a relative path.
 
+### Rewriting a script shebang
+
+Some formulae install executable scripts written in an interpreted language such as Python or Perl. Homebrew provides a `rewrite_shebang` method to rewrite the shebang of a script. This replaces a script's original interpreter path with the one installed by Homebrew and guarantees that the correct interpreter is used at execution time.
```

Comment:
> This replaces a script's original interpreter path with the one installed by Homebrew

I'm not sure this is correct? It depends on the relevant OS, I think? CC @Bo98

Reply:
Ah, yes. It's not entirely accurate. I think it depends on the dependencies a formula declares.

Reply:
For Python it's right since we don't allow system Python. For Perl, it is indeed not correct as if `uses_from_macos "perl"` is used then it will use system Perl as the shebang on macOS.

It's perhaps also worth clarifying that `rewrite_shebang` is only needed where build system doesn't already do it (it's not needed under many `pip` and `ExtUtils::MakeMaker` cases).